Tests/MongoSwiftTests/UnifiedTestRunner/UnifiedTestRunner.swift

```diff
@@ -175,46 +188,41 @@ struct UnifiedTestRunner {
                     entities: try file.createEntities?.toEntityMap() ?? [:],
                     internalClient: self.internalClient
                 )
-
                 // Workaround for SERVER-39704:  a test runners MUST execute a non-transactional distinct command on
                 // each mongos server before running any test that might execute distinct within a transaction. To ease
                 // the implementation, test runners MAY execute distinct before every test.
                 if self.topologyType.isSharded && !MongoSwiftTestCase.serverless {
                     let collEntities = context.entities.values.compactMap { try? $0.asCollection() }
-                    for (_, client) in try self.internalClient.asMongosClients() {
+                    let clientMap = try await self.internalClient.asMongosClients()
+                    for (_, client) in clientMap {
                         for entity in collEntities {
-                            _ = try client.db(entity.namespace.db).runCommand(
+                            _ = try? await client.db(entity.namespace.db).runCommand(
```

Comment:
it seems like a sign to me that something else is amiss if we need to `try?` here. I don't see why the method being async would affect whether or not the call to distinct here succeeds. could you explain why you added this?

Reply:
hmmm, I'd still like to understand why this only shows up now and never came up when the runner was sync. it might point to some subtle behavioral difference we're missing that results from this refactor. I'd think that we should be doing the same operation in the same order we were before. could you maybe link me to an example evergreen run where this happens so I can see the output?

Reply:
If you go back to regular `try` do you still see the error locally and/or on Evergreen? If so, which tests trigger it?

It's possible you were only seeing this locally because you were running a subset of tests that didn't happen to create the database used in the failing test. Or rather, running all the tests only happened to work with `try` previously because the database in question happened to be created in an earlier test.

If this is in fact what is going on, then I think using `try?` here might make sense, because if the database doesn't exist then there's nothing to do. However, we do need to verify that nothing else is amiss here before committing to that.

Reply:
Replacing with `try` locally and testing on `UnifiedTests` (the problematic file I was dealing with earlier) results in an error for `Unsupported command` as part of `valid-fail`. (running a replica set sharded cluster). The msg is "NameSpaceNotFound: database operation-failure not found"

On Evergreen, the same error occurs with the same test on the sharded clusters only.

It seems like since the errors occur on `valid-fail` which just tests using random operations/entities to ensure they fail that `try?` should be okay here?